snowflake/pkg/infra/stack.go

```diff
@@ -284,6 +284,19 @@ func declareSnowflakeDatabase(
 			return nil, err
 		}
 
+		// a bit of defensive programming: explicit dependency on ingestionStage may not be strictly required
+		fileFormat, err := snowflake.NewFileFormat(ctx, "antrea-sf-flows-format", &snowflake.FileFormatArgs{
+			Database:   db.ID(),
+			Schema:     schema.Name,
+			Name:       pulumi.String(flowFileFormatName),
+			FormatType: pulumi.String("CSV"),
+			// 0x27 is the hex representation of single quote. We use it to enclose Pod labels string.
+			FieldOptionallyEnclosedBy: pulumi.StringPtr("0x27"),
+		}, pulumi.Parent(schema), pulumi.DependsOn([]pulumi.Resource{ingestionStage, dbMigrations}), pulumi.DeleteBeforeReplace(true))
```

Comment:
I don't think you need the `pulumi.DependsOn` property here so I suggest removing it altogether to avoid confusion.
The only dependency is on the `schema` and that's already ensured by the `Schema` arg.

Reply:
I didn't add any dependencies at first, but encountered an error when creating the pipe said `File format 'FLOWFORMAT' does not exist or not authorized.`, althrough previous log has shown `FileFormat antrea-sf-flows-format created`. After testing serveral rounds, I found adding these two dependencies could solve this issue.
I agree this looks confusing, do you have any suggestions I could try?

Reply:
It's a bit of a shot in the dark but first, I'd recommend using the full file format name ("<db>.<schema>.<name>") in the copy statement. I think that by adding these dependencies, you may be affecting session parameters and hiding a bug. Just like I am using the full name for the stage I think you need the full stage for the file format.

Reply:
Thanks for advices! I just tested out the simpler way (adding that option in copy into statement directly). Think twice I agree it would be better to have a custom file format if we have multiple customizations to do in the future. For now it may be a little over engineering.

Reply:
> I agree it would be better to have a custom file format if we have multiple customizations to do in the future. For now it may be a little over engineering.

The thing is we have to keep the upgrade process robust. Pulumi is supposed to help with that, but the Snowflake provider is not always perfect. Switching from inline options to separate file format may be a bit tricky. I'll let you decide as I think additional customizations are not that likely. I still think you should give a try to the full name solution, it may just fix the issue.

Reply:
Yes the full name solution solves that issue. Sorry I only edited my last comment and didn't send a new one.